common/src/main/java/edu/snu/nemo/common/ContextImpl.java

```diff
@@ -18,23 +18,36 @@
 import edu.snu.nemo.common.ir.vertex.transform.Transform;
 
 import java.util.Map;
+import java.util.Optional;
 
 /**
  * Transform Context Implementation.
  */
 public final class ContextImpl implements Transform.Context {
   private final Map sideInputs;
+  private Optional<String> data;
```

Comment:
How about letting data `Nullable` and returning `Optional.ofNullable(data)` in `getSerializedData`?

Reply:
@seojangho I've addressed your comment :)